ozaria/site/components/interactive/PageInteractive/draggableOrdering/index.vue

```diff
@@ -61,6 +72,59 @@
         }
 
         return undefined
+      },
+
+      userAnswer () {
+        return this.promptSlots.map((s) => s.id)
+      },
+
+      solutionCorrect () {
+        for (let i = 0; i < this.userAnswer.length; i++) {
+          if (this.userAnswer[i] !== this.localizedInteractiveConfig.solution[i]) {
+            return false
+          }
+        }
+
+        return true
+      }
+    },
+
+    methods: {
+      async submitSolution () {
+        this.showModal = true
+        this.submitEnabled = false
+
+        // TODO save through vuex and block progress until save is successful
+        // TODO fix submission format
+        await putSession(this.interactive._id, {
+          json: {
+            codeLanguage: this.codeLanguage,
+            submission: {
+              correct: this.solutionCorrect,
+              submittedSolution: this.userAnswer
+            }
+          }
+        })
+      },
+
+      closeModal () {
+        if (this.solutionCorrect) {
+          this.$emit('completed')
+        } else {
+          this.resetAnswer()
+        }
+
+        this.showModal = false
+        this.submitEnabled = true
+      },
+
+      resetAnswer () {
+        // TODO consolidate with initial state setting
```

Comment:
TODO pending?

Reply:
Same for all 3 types of interactives

Reply:
These are resolved in answer randomization PR

Reply:
Are we going to support the randomization? As far as I remember, we were going to punt that?

Reply:
The last conversation I was a part of for this was that we were going to try to do it in 15 minutes and if we could then we'd launch with it.    I took a stab earlier this week as I was starting UX work and it was straightforward https://github.com/codecombat/codecombat/pull/5478

Reply:
Oh ok, that sounds good if it is straightforward, just make sure that Maya is aware of this.